src/csharp/Grpc.Core/GrpcEnvironment.cs

```diff
@@ -238,6 +240,21 @@ public static void SetHandlerInlining(bool inlineHandlers)
         }
 
         /// <summary>
+        /// Occurs when <c>GrpcEnvironment</c> is about the start the shutdown logic.
+        /// </summary>
+        public static event EventHandler ShuttingDown
```

Comment:
It would be better to use a field like event here:

```csharp
public static event EventHandler ShuttingDown;
```

That would perform the subscribe/unsubscribe aspects in a thread-safe way - this doesn't at the moment.